internal/controller/promotion/helm.go

```diff
@@ -98,6 +102,9 @@ func (h *helmer) apply(
 		if err = h.setStringsInYAMLFileFn(chartYAMLPath, changes); err != nil {
 			return nil, fmt.Errorf("error updating dependencies for chart %q: %w", chart, err)
 		}
+		if err = h.prepareDependencyCredentialsFn(context.TODO(), homeDir, chartYAMLPath, namespace); err != nil {
+			return nil, fmt.Errorf("error resolving credentials for chart %q: :%w", chart, err)
```

Comment:
nit: I think `error` prefix can be duplicated in outer context. How about removing it? (Ref - [Uber go style guide - error wrapping](https://github.com/uber-go/guide/blob/master/style.md#error-wrapping)
```suggestion
			return nil, fmt.Errorf("resolve credentials for chart %q: :%w", chart, err)
```

Reply:
Ah, the Uber go style guide is where this is coming from. I wondered about it before, after observing this style of errors in other places in the code base.

I agree it's beneficial here (although in _some_ other places, it could use more context to answer _why_ questions specific to the returned error).

I will make some adjustments, thanks.